src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -168,7 +168,7 @@ public override bool Equals(object obj)
                 return false;
 
             var value = (StackValue)obj;
-            return this.Kind == value.Kind && this.Flags == value.Flags && this.Type == value.Type;
+            return this.Kind == value.Kind && this.IsReadOnly == value.IsReadOnly && this.Type == value.Type;
```

Comment:
It may be better to inline the subset to compare into the calller.

I think that the right set of flags to compare will depend on who is calling it, and that there is no right subset to compare to make everybody happy.

Reply:
I changed this to only check for the readonly flag, since PEVerify overloaded the `==` in the same way. I think it was implemented this way, because comparing the equality of two stack values by also considering flags like `PermanentHome` or `ThisPtr` is never actually used.

Anyway, I do agree that it would make more sense semantically to compare all flags in the equality operations and inline checks that require a subset.